internal/lock/lockfile.go

```diff
@@ -106,10 +106,8 @@ func (f *File) Save() error {
 	// such as `devbox update` or `devbox add` or `devbox remove`.
 	for pkgName, pkg := range f.Packages {
 		for sys, sysInfo := range pkg.Systems {
-			if !isDirty && sysInfo.StorePath != "" {
+			if !isDirty && sysInfo.outputIsFromStorePath {
```

Comment:
Can this be just `!isDirty` ? 

Then we can get rid of `outputIsFromStorePath`

Reply:
Actually is isDirty even working as expected? I would think isDirty is always true when loading a legacy lockfile (because we add the outputs).